P/PETSc/build_tarballs.jl

```diff
@@ -1,23 +1,26 @@
+# PETSc 3.21.5 with OpenBLAS and static compilations of SuperLU_Dist, SuiteSparse, MUMPS, Hypre, triangle and TetGen on machines that support it
 using BinaryBuilder, Pkg
 using Base.BinaryPlatforms
 const YGGDRASIL_DIR = "../.."
 include(joinpath(YGGDRASIL_DIR, "platforms", "mpi.jl"))
 
 name = "PETSc"
-version = v"3.20.5"
-petsc_version = v"3.20.5"
+version = v"3.21.5"
+petsc_version = v"3.21.5"
 
-MPItrampoline_compat_version="5.2.1"
+MPItrampoline_compat_version="~5.5.0"
 MicrosoftMPI_compat_version="~10.1.4" 
 MPICH_compat_version="~4.1.2"    
```

Comment:
Why the tilde compat?  I don't think these packages are known to break API/ABI in minor versions, are they?

Reply:
perhaps it is different for `MPItrampoline`, but I certainly had runtime errors with `MPICH` when the minor version was changed.